### DISCUSSION TITLE
useSuspenseInfiniteQuery triggers loading state when fetching additional pages

### DISCUSSION THREAD

**Discussion by withintheruins14** [TanStack/query#8178] [Q&A]
I've implemented an infinite scroll where the first page is prefetched on the server and `useSuspenseInfiniteQuery` is used in the client to read the prefetched data. It works for the initial load but when the infinite scroll fetches the next page the suspense boundary enters a loading state - it's weird because the table turns into a spinner. How can I prevent the suspense boundary from doing this after the initial data is present? Thanks!

**Comment by withintheruins14** [comment]
hmm, maybe I need `startTransition `?

**Reply by TkDodo** [reply]
yes, that or `useDeferredValue`. It’s “just” a react thing, not specific to React Query.
